code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -632,12 +632,16 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 		if(inv)
 			inv.update_icon()
 
+	var/hidden = (slot_gloves in check_obscured_slots())
```

Comment:
Prob good to refactor this one day to be calculated just once. And not for each clothing piece.
Given the size of this refactor, I consider it outside of the scope of this PR